Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift

```diff
@@ -36,33 +40,46 @@ internal class ExperienceRenderer: ExperienceRendering {
         self.analyticsObserver = ExperienceStateMachine.AnalyticsObserver(container: container)
     }
 
-    func show(experience: Experience, priority: RenderPriority, published: Bool, completion: ((Result<Void, Error>) -> Void)?) {
+    func show(experience: ExperienceData, completion: ((Result<Void, Error>) -> Void)?) {
         guard Thread.isMainThread else {
             DispatchQueue.main.async {
-                self.show(experience: experience, priority: priority, published: published, completion: completion)
+                self.show(experience: experience, completion: completion)
             }
             return
         }
 
-        if priority == .normal && stateMachine.state != .idling {
+        if !(experience.experiment?.shouldExecute() ?? true) {
+            // if we get here, it means we did have an experiment, but it was the control group and
+            // we should not continue. So track experiment_entered analytics for it (always)..
+            experience.experiment?.track(appcues: appcues, experimentID: experience.experimentID)
+            // and exit early
+            completion?(.failure(ExperienceRendererError.experimentControl))
+            return
+        }
+
+        if experience.priority == .normal && stateMachine.state != .idling {
             dismissCurrentExperience(markComplete: false) { result in
                 switch result {
                 case .success:
-                    self.show(experience: experience, priority: priority, published: published, completion: completion)
+                    self.show(experience: experience, completion: completion)
                 case .failure(let error):
                     completion?(.failure(error))
                 }
             }
             return
         }
 
+        // if we get here, either we did not have an experiment, or it is active and did not exit early (not control group).
+        // if an active experiment does exist, it should now track the experiment_entered analytic
+        experience.experiment?.track(appcues: appcues, experimentID: experience.experimentID)
```

Comment:
this case was not tracked above on line 54, to avoid double tracking in the case of recursive call on line 64 during `dismissCurrentExperience`